dask/dataframe/tests/test_dataframe.py

```diff
@@ -306,19 +306,94 @@ def test_describe(method, test_values):
     assert_eq(df.describe(), ddf.describe(split_every=2, percentiles_method=method))
 
 
-@pytest.mark.parametrize('method', ['tdigest', 'dask'])
-def test_describe_empty(method):
-    if method == 'tdigest':
-        pytest.importorskip('crick')
-    # https://github.com/dask/dask/issues/2326
-    ddf = dd.from_pandas(pd.DataFrame({"A": ['a', 'b']}), 2)
-    with pytest.raises(ValueError) as rec:
-        ddf.describe(percentiles_method=method)
-    assert 'DataFrame contains only non-numeric data.' in str(rec)
+@pytest.mark.parametrize('include,exclude,percentiles', [
+    (None, None, None),
+    ('all', None, None),
+    (['number'], None, [0.25, 0.5]),
+    ([np.timedelta64], None, None),
+    (['number', 'object'], None, [0.25, 0.75]),
+    (None, ['number', 'object'], None),
+    (['object', 'datetime', 'bool'], None, None)
+])
+def test_describe(include, exclude, percentiles):
+    data = {
+        'a': ["aaa", "bbb", "bbb", None, None, "zzz"] * 2,
+        'c': [None, 0, 1, 2, 3, 4] * 2,
+        'd': [None, 0, 1] * 4,
+        'e': [pd.Timestamp('2017-05-09 00:00:00.006000'),
+              pd.Timestamp('2017-05-09 00:00:00.006000'),
+              pd.Timestamp('2017-05-09 07:56:23.858694'),
+              pd.Timestamp('2017-05-09 05:59:58.938999'),
+              None,
+              None] * 2,
+        'f': [np.timedelta64(3, 'D'),
+              np.timedelta64(1, 'D'),
+              None,
+              None,
+              np.timedelta64(3, 'D'),
+              np.timedelta64(1, 'D')] * 2,
+        'g': [True, False, True] * 4
+    }
+
+    # Arrange
+    df = pd.DataFrame(data)
+    ddf = dd.from_pandas(df, 2)
+
+    # Act
+    desc_ddf = ddf.describe(include=include, exclude=exclude, percentiles=percentiles)
+    desc_df = df.describe(include=include, exclude=exclude, percentiles=percentiles)
+    desc_ddf_computed = desc_ddf.compute()
+
+    # TODO: for timedelta columns there's overflow in var function, see #4233
+    # causing std to be nan, ignoring this cell
+    if 'f' in desc_ddf_computed:
+        desc_df['f']['std'] = None
+        desc_ddf_computed['f']['std'] = None
+
+    # Assert
+    assert_eq(desc_df, desc_ddf_computed)
+
+    # Check series
+    for col in ['a', 'c', 'e', 'g']:
+        assert_eq(
+            df[col].describe(include=include, exclude=exclude),
+            ddf[col].describe(include=include, exclude=exclude).compute())
```

Comment:
Normally we pass an uncomputed dask object to `assert_eq`. This assertion not only checks that its computed version matches the desired result (e.g. the pandas dataframe), but also checks that the various bits of metadata on the dask object are correct. You likely want to drop the `.compute()` call here and elsewhere in your tests.

Reply:
@jcrist thank you for noticing this.
I have a question regarding meta for empty dataframe. Pandas describe() output has different dtypes when frame is empty:
```>df = pd.DataFrame({'a':[2.5, 1], 'b':[True, False]})
>df.describe(include='all').dtypes
a    float64
b     object
dtype: object
>df_none = pd.DataFrame({'a':[None,None], 'b':[None,None]})
>df_none.describe(include='all').dtypes
a    int64
b    int64
dtype: object
```
meta is being assigned at the moment when it's unknown whether dataframe is empty, therefore pandas output and dask dataframe meta might not match. Could you suggest the right way to handle such case?

Reply:
This looks like a bug in pandas. Ideally the output shape would depend only on the dtypes, not the values. I'll verify and open a pandas issue.

I haven't looked at the implementation, but can you use `self._meta_nonempty` to compute the metadata rather than `self._meta`?

Reply:
We'll see what comes of https://github.com/pandas-dev/pandas/issues/26397. It's not clear to me what's most useful to users.

Reply:
>can you use self._meta_nonempty to compute the metadata rather than self._meta

@TomAugspurger thank you for suggestion, sure I should've used _meta_nonempty.

I think empty describe has different dtypes because it doesn't know dtypes of columns and cannot calculate any stats except count and unique which are of integer type.
```
> df_none.describe(include='all')
        a  b
count   0  0
unique  0  0
> df.describe(include='all')
              a     b
count   2.00000     2
unique      NaN     2
top         NaN  True
freq        NaN     1
mean    1.75000   NaN
std     1.06066   NaN
min     1.00000   NaN
25%     1.37500   NaN
50%     1.75000   NaN
75%     2.12500   NaN
max     2.50000   NaN
```